src/main/java/teammates/common/util/Config.java

```diff
@@ -103,53 +94,121 @@ public final class Config {
-        EMAIL_SERVICE = properties.getProperty("app.email.service");
-        SENDGRID_APIKEY = properties.getProperty("app.sendgrid.apikey");
-        MAILGUN_APIKEY = properties.getProperty("app.mailgun.apikey");
-        MAILGUN_DOMAINNAME = properties.getProperty("app.mailgun.domainname");
-        MAILJET_APIKEY = properties.getProperty("app.mailjet.apikey");
-        MAILJET_SECRETKEY = properties.getProperty("app.mailjet.secretkey");
-        SEARCH_SERVICE_HOST = properties.getProperty("app.search.service.host");
-        ENABLE_DATASTORE_BACKUP = Boolean.parseBoolean(properties.getProperty("app.enable.datastore.backup", "false"));
-        MAINTENANCE = Boolean.parseBoolean(properties.getProperty("app.maintenance", "false"));
+                Arrays.asList(getProperty(properties, devProperties, "app.maintainers", "").split(",")));
+        SUPPORT_EMAIL = getProperty(properties, devProperties, "app.crashreport.email");
+        EMAIL_SENDEREMAIL = getProperty(properties, devProperties, "app.email.senderemail");
+        EMAIL_SENDERNAME = getProperty(properties, devProperties, "app.email.sendername");
+        EMAIL_REPLYTO = getProperty(properties, devProperties, "app.email.replyto");
+        EMAIL_SERVICE = getProperty(properties, devProperties, "app.email.service");
+        SENDGRID_APIKEY = getProperty(properties, devProperties, "app.sendgrid.apikey");
+        MAILGUN_APIKEY = getProperty(properties, devProperties, "app.mailgun.apikey");
+        MAILGUN_DOMAINNAME = getProperty(properties, devProperties, "app.mailgun.domainname");
+        MAILJET_APIKEY = getProperty(properties, devProperties, "app.mailjet.apikey");
+        MAILJET_SECRETKEY = getProperty(properties, devProperties, "app.mailjet.secretkey");
+        SEARCH_SERVICE_HOST = getProperty(properties, devProperties, "app.search.service.host");
+        ENABLE_DATASTORE_BACKUP = Boolean.parseBoolean(
+                getProperty(properties, devProperties, "app.enable.datastore.backup", "false"));
+        MAINTENANCE = Boolean.parseBoolean(getProperty(properties, devProperties, "app.maintenance", "false"));
+
+        // The following properties are not used in production server.
+        // So they will only be read from build-dev.properties file.
+        APP_LOCALDATASTORE_PORT = Integer.parseInt(devProperties.getProperty("app.localdatastore.port", "8484"));
+        ENABLE_DEVSERVER_LOGIN = Boolean.parseBoolean(devProperties.getProperty("app.enable.devserver.login", "false"));
+        TASKQUEUE_ACTIVE = Boolean.parseBoolean(devProperties.getProperty("app.taskqueue.active", "true"));
     }
 
     private Config() {
         // access static fields directly
     }
 
-    static String getBaseAppUrl() {
-        return isDevServer() ? "http://localhost:" + getPort() : "https://" + APP_ID + ".appspot.com";
+    /**
+     * Returns the a default frontend URL if it is not set in property file(s).
+     */
+    public static String getDefaultFrontEndUrl() {
```

Comment:
This can be package level visibility instead of public?

Reply:
Oh yes. That makes sense. Thanks for pointing that out!